java-generator/maven-plugin/src/main/java/io/fabric8/java/generator/maven/plugin/JavaGeneratorMojo.java

```diff
@@ -42,9 +42,31 @@
     @Parameter(property = "fabric8.java-generator.enum-uppercase", required = false)
     Boolean enumUppercase = null;
 
+    @Parameter(property = "fabric8.java-generator.prefix-strategy", required = false)
+    Config.Prefix prefixStrategy = null;
+
+    @Parameter(property = "fabric8.java-generator.suffix-strategy", required = false)
+    Config.Suffix suffixStrategy = null;
+
+    @Parameter(property = "fabric8.java-generator.always-preserve-unknown", required = false)
+    Boolean alwaysPreserveUnknown = null;
+
+    @Parameter(property = "fabric8.java-generator.extra-annotations", required = false)
+    Boolean extraAnnotations = null;
+
+    @Parameter(property = "fabric8.java-generator.code-structure", required = false)
```

Comment:
How do we hide some configurations?

Reply:
What do you mean by "hide"?

Reply:
Like we do in the CLI:
https://github.com/fabric8io/kubernetes-client/pull/3817/files#diff-1c647fc1e9204b845f5024b4ad0a74a5d58ded1e458286d577a163c1ab8d5dd1R67

It's an available option but is removed from documentation etc.

Reply:
I'm not sure there's an equivalent option for Maven plugins.
You either want to expose the option or not. There is also no help CLI message generated (AFAIK).

Reply:
OK; maybe I'm wrong:
```shell
mvn help:describe -Dplugin=$pluginName -Dmojo=$pluginMojo -Dfull=true
```

But if you're doing this, you probably want to see all of the parameters.

Reply:
Can confirm I can see all the parameters using:
```
mvn help:describe -Dplugin=io.fabric8:java-generator-maven-plugin -Dmojo=io.fabric8.java.generator.maven.plugin.JavaGeneratorMojo -Dfull=true -Ddetail 
```

Unfortunately I cannot find any way to hide one, should we add a follow-up issue?
Should we name the "hidden" properties as "unsupported" or something?

Reply:
If the reasoning is because these options are not supported (what we talked about), I would just prefix them with `preview`, or something like this.

```suggestion
    @Parameter(property = "fabric8.java-generator.preview.code-structure", required = false)
    Config.Prefix previewPrefixStrategy = null;
```

Reply:
Using the [`readonly` property](https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Parameter.html#readonly--) seems to do job @manusa , and the integration tests are still working, so the user-defined properties are actually taken into account.

Reply:
ouch @manusa , concurrent update on the same subject ☝️  🙂